src/common/com/intellij/plugins/haxe/model/type/HaxeTypeResolver.java

```diff
@@ -408,10 +405,33 @@ static public ResultHolder getTypeFromType(@NotNull HaxeType type, @Nullable Hax
         references.add(partResult);
       }
     } else if (null != resolvedHaxeClass) {
-      // A bare type declaration uses the default types.
-      // TODO: Add types with constraints, if known, or "unknown" for all declared paratmers here.
+
+      PsiElement resolved = expression.resolve();
```

Comment:
Why are we re-resolving here?  Isn't resolvedHaxeClass the PSI element that we end up with after resolve()?

Reply:
i could use 
```java
ResultHolder[] referenceArray= expression.resolveHaxeClass()
						.getGenericResolver()
						.getSpecificsFor(resolvedHaxeClass);
```
But then i need to  either accept that all error messages with type mismatch for dynamic will show as `Dynamic<Dynamic>`
```xml
 <error descr="Incompatible type: String should be Class<Dynamic<Dynamic>>">
var dynamicClass6:Class<Dynamic> = "" ;                             
```
 or i need to figure out a way to prevent `Dynamic` from being resolved as `Dynamic<Dynamic>`
 I guess the later option is the best, as the hack mentioned above could be dropped as well,  but i could not figure out how to prevent it as the  class/model for `Dynamic`  is `Dynamic<T> its just the compiler that also accepts just `Dynamic`
 Other legal values  could for instance be `Dynamic<String>`.
 ```haxe
 @:coreType @:runtimeValue abstract Dynamic<T> {}
 ```

Reply:
For the "Dynamic<Dynamic>" printing, we should probably just fix the pretty-printer (I think that's HaxeTypeModel.getPresentationString()) to not add the type parameter in that specific case.  I think we would want to show the type parameter if it was "Dynamic<String>".  OTOH, it's a good clue to us as to why type parameter issues are failing. :D